packages/server/src/routes/grants.js

```diff
@@ -431,15 +431,33 @@ router.put('/:grantId/interested/:agencyId', requireUser, async (req, res) => {
         return;
     }
 
-    await db.markGrantAsInterested({
-        grantId,
-        agencyId,
-        userId: user.id,
-        interestedCode,
-    });
-
-    const interestedAgencies = await db.getInterestedAgencies({ grantIds: [grantId], tenantId: user.tenant_id });
-    res.json(interestedAgencies);
+    try {
+        // Calling the existing function to mark the grant as interested
+        await db.markGrantAsInterested({
+            grantId,
+            agencyId,
+            userId: user.id,
+            interestedCode,
+        });
+        // Query to check if the interestedCode corresponds to 'Interested'
+        const interestedStatus = await knex('interested_codes')
+            .select('status_code')
+            .where({ id: interestedCode })
+            .first();
+
+        // Follow  or Unfollow the grant based on the interestedCode
+        if (interestedStatus?.status_code === 'Interested') {
+            await followGrant(knex, grantId, user.id);
+        } else {
+            await unfollowGrant(knex, grantId, user.id);
+        }
```

Comment:
This should come before the call to `db.markGrantAsInterested()`, and be wrapped in a transaction so that we can roll back in case `db.markGrantAsInterested()` fails:
```js
// Follow  or Unfollow the grant based on the interestedCode
const trx = await knex.transaction();
if (interestedStatus?.status_code === 'Interested') {
    await followGrant(trx, grantId, user.id);
} else {
    await unfollowGrant(trx, grantId, user.id);
}
try {
    // Calling the existing function to mark the grant as interested
    await db.markGrantAsInterested({
        grantId,
        agencyId,
        userId: user.id,
        interestedCode,
    });
} catch (error) {
    // Roll back the follow/unfollow operation since marking grant as interested failed
    trx.rollback();
    console.error('Error in marking as interested:', error);
    throw error;
}
// Commit the follow/unfollow operation since everything succeeded
trx.commit();
```

Note: The order of `db.markGrantAsInterested()` and either `followGrant()` or `unfollowGrant()` wouldn't matter if we were able to pass the transaction to `db.markGrantAsInterested()`, since that would allow us to commit or rollback as a unit. Modifying that function is arguably out of scope for this issue, but just to illustrate what I mean:
```js
async function markGrantAsInterested({
    trx = knex, grantId, agencyId, userId, interestedCode,
}) {
    const results = await trx(TABLES.grants_interested)
        .insert({
            agency_id: agencyId,
            grant_id: grantId,
            user_id: userId,
            interested_code_id: +interestedCode,
        });
    return results;
}
```

Reply:
@TylerHendrickson adding transaction to rollback if markGrantAsInterested fails is a great suggestion, thanks for the detailed implementation. I have incorporated your suggestion, I will push the code with the suggested changes.

Reply:
@TylerHendrickson I have added await to the transaction.commit() and transaction.rollback().